frame/contracts/rpc/src/lib.rs

```diff
@@ -259,7 +259,7 @@ where
 		} = instantiate_request;
 
 		let value: Balance = decode_hex(value, "balance")?;
-		let gas_limit: Weight = decode_hex(gas_limit, "weight")?;
+		let gas_limit = decode_hex(gas_limit, "weight")?;
```

Comment:
Sooooo what type is this being decoded to? I mean it's likely that the compiler's type inference algorithm had figured this out, but I suspect the reason why we'd still put a type annotation because we'd like to be explicit here.

Reply:
made explicit